### DISCUSSION TITLE
Removing document raises below exception

### DISCUSSION THREAD

**Discussion by DevDeepakJoshi** [dotnet/roslyn#61241] [Q&A] [answered]
I am trying remove a document from workspace but i run into this issue once i call **TryApplyChanges** on **workspace**. This is the error message i get

**System.AggregateException: 'One or more errors occurred. (Cannot modify an evaluated object originating in an imported file "C:\Program Files\dotnet\sdk\6.0.101\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.DefaultItems.props".)'**

 I saw a similar thread on [Stackoverflow ](https://stackoverflow.com/questions/47759997/getting-error-cannot-modify-an-evaluated-object-originating-in-an-imported-file/47780992#47780992) but my issue still persists even after trying those changes.

Any idea how i can resolve this issue,.

Thanks in advance.

**Comment by CyrusNajmabadi** [comment]
Can you please use the "report a problem" feature in VS to report this?  It should help us get more data on what went wrong. 

Thanks!

**Reply by DevDeepakJoshi** [reply]
Hi CyruNajmabadi,

I was able to fix this issue. 

With both the nodes added in the project file.
`<ItemGroup>
    <Compile Include="C:\workdir\Override tests\Configurator\Devspace\DefaultUser\TestModel\Code\Model\XYZ.cs" />
  </ItemGroup>
  <PropertyGroup>
    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
  </PropertyGroup>`

If these nodes are present, RemoveDocument work fine and TryApplyChanges does not raise any exception.
I'll close this discussion.

**Comment by DevDeepakJoshi** [comment, marked as answer]
If ItemGroup for file name is present and EnableDefaultCompileItems set to false, RemoveDocument works fine.

e.g.
  `<ItemGroup>
    <Compile Include="C:\workdir\Override tests\Configurator\Devspace\DefaultUser\TestModel\Code\Model\XYZ.cs" />
  </ItemGroup>
  <PropertyGroup>
    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
  </PropertyGroup>`

**Reply by jasonmalinowski** [reply]
@DevDeepakJoshi It sounds though like we may have a bug here? You shouldn't have to tweak your project in that way I don't think, but unless I'm missing something further here.
